### ISSUE TITLE
Ignore `EPIPE` on writable streams

### ISSUE THREAD

**Issue by ehmicky** [sindresorhus/execa#823]
When piping two processes, the source process usually ends first, closes the destination process' `stdin`, then the destination ends.

However, there are some situations when the destination process decides when to end instead. This usually happens when either: 
  - The source process pipes some infinite stream of data to the destination process. For example, `yes | head -n10`
  - The source process pipes a finite stream of data, but the destination process only needs a subset of it. For example, `cat file.txt | head -n10`.

In a Unix shell: 
  - First, the destination process ends. 
  - Then, on the next write system call the source process does, it receives a `SIGPIPE` signal. 
  - By default, this terminates the source process. 
  - But it can handle that signal, in which case the kernel sends back an `EPIPE` error. The source process can then proceed to do some graceful exit.

Node.js does not provide with out-of-box process piping (I am working on it for Execa! :) ) but it does provide with ways to pipe a source process `stdout` to a destination process `stdin`. What happens in the above situation is: 
  - First, the destination process closes its `stdin`.
  - Then, the next time the source process `stdout` writes to the destination process `stdin`, it receives an `EPIPE` error.
  - This makes the destination process `stdin` error. 
  - This does not make the source process `stdout` error, but it makes it stop reading due to the `unpipe` event.

`EPIPE` seems to be the only way to handle the above situation. In particular, there seems to be no way to know whether the next write call to the destination process `stdin` will result in an `EPIPE` or not. `EPIPE` fails before any event is emitted or any property is set on both the destination `childProcess` and its `childProcess.stdin` stream. For example, `childProcess.stdin.writable` is `true` before that call. 

In other words, it seems that, by design, Node.js considers that `EPIPE` error (which is returned by the underlying `Writable._write()` method) to be the "event" we should listen to in order to know piping is done (in a situation where the destination process decides when to end piping).

This results in the following: `EPIPE` errors should be ignored on `childProcess.stdin`. Moreover, it should be ignored on any writable stream Execa uses, such as any passed to the `stdout` option since those could be `{stdout: childProcess.stdin}`. The stream would still end on `EPIPE`, but this would not be reported as an error.
